### PR TITLE
Refactor errors and fsx files

### PR DESCRIPTION
**Pull request by guillermooo**
Better support for .fsx files.

Some refactoring around errors received from fsac, bug fixes, etc.

### CODE REVIEW

**Comment by guillermooo** [comment]
@rojepp  PTAL commit by commit. Sorry, I got carried away with the changes and lumped everything together.

File: FSharp/lib/editor.py [hunk 1]
```diff
@@ -56,6 +79,9 @@ def refresh(self, fs_file):
         assert isinstance(fs_file, FSharpFile), 'wrong argument: %s' % fs_file
         # todo: run in alternate thread
 
+        if fs_file.is_script_file:
+            return
+
```
Comment (rojepp) [inline FSharp/lib/editor.py]: Could this method (`refresh`) be better named to indicate that it sets active project? And in the case of a script file or the case that there is no project file for a .fs file, shouldn't we 'unset' the project in fsac so we don't get wrong results?

File: FSharp/lib/project.py [hunk 2]
```diff
@@ -45,20 +45,31 @@ def is_fsharp_file(self):
...
     def is_script_file(self):
-        return (extension_equals(self.view_or_fname, '.fsx') or
-                extension_equals(self.view_or_fname, '.fsi'))
+        '''
+        Returns `True` if `self` is a .fsx/.fsi/.fsscript file.
+        '''
+        return (extension_equals(self.path, '.fsx')
+                or extension_equals(self.path, '.fsi')
+                or extension_equals(self.path, '.fsscript')
+                )
```
Comment (rojepp) [inline FSharp/lib/project.py]: I commented on individual commit instead of here. Why .fsscript? That extension is new to me.

**Comment by rojepp** [comment]
I see squigglies in .fsx files now, great! 

There's code in here that seem to implement tooltips, but I don't see any?

**Comment by rneatherway** [comment]
Just a note, if it wasn't clear, you can now load multiple projects and they will all be 'active', you aren't limited to just one at a time anymore. So load away and let FSAC worry about it!

**Comment by rojepp** [comment]
@rneatherway How can you 'unset' a project? Maybe it isn't necessary, I'm not sure. If you have previously opened a project, and then navigate to an .fsx file, you wouldn't wan't fsac to analyze it as if it is part of the project. I'm thinking that could lead to errors *not* being caught by FSAC. Like missing `#r ...` statements for instance.

**Comment by rneatherway** [comment]
You can't unset a project, I don't think it's needed. .fsx files are always analysed completely independent of any project, which is consistent with Visual Studio.

File: FSharp/lib/editor.py (hunk 1 above)
Comment (guillermooo) [inline FSharp/lib/editor.py]: Will do.

**Comment by guillermooo** [comment]
Ah, ok, didn't know that. Then I'll make sure not to unload projects. Wondered the same as @rojepp.

**Comment by guillermooo** [comment]
@rojepp Re. tooltips: Press Ctrl+k, Ctrl+i on any symbol.

(Often, I get multiple "ERROR NOT PARSED" from fsac before the tooltip finally shows up.)

**Comment by rneatherway** [comment]
You must send a parse command for a file before anything else, like
requesting tooltips. This is just making fsac aware of the current state of
the file. I would do it before each other command, and on idle for errors.

**Comment by guillermooo** [comment]
@rneatherway I'm currently re-parsing on_idle after 1,5 secs no matter what.

https://github.com/fsharp/sublime-fsharp-package/blob/4abd84c5c8c69f94cac90132f10527906c37686e/FSharp/xevents.py#L37

I get the error I mentioned even with clean views, so I'm not sure what may be wrong.

**Comment by rneatherway** [comment]
Do you parse on opening a file, and before each other command?

For example, the emacs code for goto definition: https://github.com/fsharp/emacs-fsharp-mode/blob/master/fsharp-mode-completion.el#L457

**Comment by rneatherway** [comment]
If you do, then can you post a log of the communication?

**Comment by guillermooo** [comment]
I do parse on_load. I'll experiment with re-parsing before every command and see if that helps.

**Comment by guillermooo** [comment]
@rneatherway @rojepp 

Ok, forcing a re-parse does seem to help. PTAL at the latest changes. If everything's ok, I'll merge with master.

**Comment by rojepp** [comment]
LGTM! 
Great to see progress here!
I'm sure there are still issues to fix but this is progress.
It seems the re-parse makes completions usable. Please re-enable fs_dot and I think this should be merged. Btw, @guillermooo can we get your contact details? Robin wanted to mail you and I've wanted to chat with you this week. :)

**Comment by guillermooo** [comment]
@rojepp Email me at the address in my profile if you want. Does that work?

**Pull request merged**